nexus/db-queries/src/db/datastore/oximeter.rs

```diff
@@ -75,6 +100,30 @@ impl DataStore {
         Ok(())
     }
 
+    /// Mark an Oximeter instance as deleted
+    ///
+    /// This method is idempotent and has no effect if called with the ID for an
+    /// already-deleted Oximeter.
+    pub async fn oximeter_delete(
+        &self,
+        opctx: &OpContext,
+        id: Uuid,
+    ) -> Result<(), Error> {
+        use db::schema::oximeter::dsl;
+
+        let now = Utc::now();
+
+        diesel::update(dsl::oximeter)
+            .filter(dsl::time_deleted.is_null())
+            .filter(dsl::id.eq(id))
+            .set(dsl::time_deleted.eq(now))
+            .execute_async(&*self.pool_connection_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(())
+    }
+
     /// List the oximeter collector instances
```

Comment:
Same note, maybe update this docstring to indicate it returns non-deleted instances.